internal/hookruntime/state.go

```
@@ -0,0 +1,413 @@
+	paths, err := CurrentUserPaths()
+	if err != nil || strings.TrimSpace(paths.Launcher) == "" {
+		return State{}, false, err
+	}
+	return readTrustedAt(paths, executable)
+}
+
+func readTrustedAt(paths Paths, executable string) (state State, recognized bool, err error) {
+	if !samePath(executable, paths.Launcher) {
+		return State{}, false, nil
+	}
+	recognized = true
```

golangci-lint (2.12.2)

[error] ineffectual assignment to recognized

(ineffassign)